enhancements/20220525-system-logging.md

```diff
@@ -0,0 +1,174 @@
+
+The default username is admin. To get the password for the admin user run the following command:
+
+```bash
+kubectl get secret --namespace cattle-logging-system loki-stack-grafana --output jsonpath="{.data.admin-password}" | base64 --decode
+```
+
+##### Log Querying
+
+Once the UI is open on the left tab you can select the "Explore" tab to take you to a page where you manually send
+queries to loki:
+
+![](./20220525-system-logging/loki-explore-tab.png)
+
+In the search bar you can enter queries to select the logs you are interested in. The query language is described in the
+[grafana loki docs](https://grafana.com/docs/loki/latest/logql/log_queries/). For example, to select all logs in the
+`cattle-logging-system`:
+
+![](./20220525-system-logging/loki-explore-query-kubernetes-namespace-name-cattle-logging-system.png)
+
+Or to select harvester host machine logs from the `rke2-server.service` service:
+
+![](./20220525-system-logging/loki-explore-query-systemd-rke2-server.png)
+
+### User Experience In Detail
+
+### API changes
+
+None.
+
+## Design
+
+### Implementation Overview
+
+The bulk of the logging functionality is handled by installing the [`rancher-logging`](https://github.com/rancher/charts/tree/dev-v2.7/charts/rancher-logging/100.1.3%2Bup3.17.7/)
+to deploy the main logging components:
+
+| Name                | Purpose                                                                    |
+|---------------------|----------------------------------------------------------------------------|
+| logging-operator    | manages the `ClusterFlow`s and `ClusterOutput`s defining log routes        |
+| fluentd             | the central log aggregator which will forward logs to other log collectors |
+| fluent-bit          | collects the logs from teh cluster pods                                    |
+| journald-aggregator | deploys a pod to collect the journal logs from each node                   |
+
+The journald-aggregator is a `fluent-bit` pod which collects the node logs by mounting the host's `/var/log/journal`
+directory to the pod. Using the [`systemd`](https://docs.fluentbit.io/manual/pipeline/inputs/systemd) input plugin, the
+logs can be filtered by the fields in the log entry (ex `_SYSTEMD_UNIT`, `_TRANSPORT`, etc). Collecting all the jogs
+from `/var/log/journal` is too much, so we only select logs from some important services: rke2-server, rke2-agent,
+rancherd, rancher-system-agent, wicked, iscsid, and kernal logs.
+
+The logging feature is enabled by default; however, due to an [issue](https://github.com/harvester/harvester/issues/2719)
+with enabling and disabling `ManagedChart`s, it cannt be disabled.
+
+### Test plan
+
+1. Install harvester with the implemented charts and CRDs
+2. Verify logs are being routed to the configured ClusterOutput
+3. Change log output settings
+4. Verify logs are routed to the newly configured ClusterOutput
+
+### Upgrade strategy
+
+No user intervention is required during the upgrade.
```

Comment:
Similar with, the `logging` should be enabled when upgrading from v1.0.3.
https://github.com/harvester/harvester/issues/2750